### ISSUE TITLE
assembler commandline parameter parsing is broken (wasm crash on "-fr", failure on "-?")

### ISSUE THREAD

**Issue by winspool** [open-watcom/open-watcom-v2#1265]
I tried a "helloword" in x86 asm, but OpenWatcom crashes:

Source from german wikipedia: 
[hw_16t.zip](https://github.com/open-watcom/open-watcom-v2/files/14873779/hw_16t.zip)
https://de.wikipedia.org/wiki/Liste_von_Hallo-Welt-Programmen/Assembler#DOS_(COM-Datei),_MASM,_TASM

My command:
```
owcc -v -bdos -mcmodel=t hw_16t.asm -o hw_16t.com
Open Watcom x86 32-bit Compilers Driver Program
Version 2.0 beta Mar 30 2024 21:49:40 (64-bit)
Copyright (c) 2002-2024 The Open Watcom Contributors. All Rights Reserved.
Portions Copyright (c) 1988-2002 Sybase, Inc. All Rights Reserved.
Source code is available under the Sybase Open Watcom Public License.
See https://github.com/open-watcom/open-watcom-v2#readme for details.
	wasm -bt=dos -ms -0 -fo=.o -fr hw_16t.asm 
Error: Compiler returned a bad status compiling 'hw_16t.asm'
```

When using `wasm` directly, a segmentation fault is detected in:
File: `split2.c`
Function: `_splitpath2`
Buggy arg:  `inp` has the value 0x00000004
Bug location visible in `wd`: https://imgur.com/sfo6gMx.png

(Scrolling (with mouse or courser keys) in the `wd` window produces garbage characters.
 That's different from bug #1263)

--- 
To try to understand the `wasm` commandline options, i tried the commandline help:
`wasm --help`

The `wasm` commandline-help is printed, but has failures:
```
-?     print this help
```
this is wrong:
`wasm -?` fails with
```
Error: Cannot Open File - No filename specified.
```
---

Another failure in the `wasm` commandline help:
```
-fr[=<file>]    set error file name
```
`owcc` adds `-fr` without a filename to the `wasm` commandline.
From the commandline help, the filename should be optional,
but `-fr` without a filename crashes `wasm`.

I tried also `wasm -fr  bld/wasmtest/test1/hello.asm` 
and `wasm` crashed as before.

---

The ctools guide for wasm has also failures:
* `--help` is not listed
* `-fr` is not listed
* `-?` and `-h` are listed as usable to print the commandline help, but both do not work

**Comment by winspool** [comment]
For the other assembler, i tested `hello.asm` for `ppc`:
```
cd bld/as/ppc/test
$ owcc -march=ppc hello.asm -c -o hello_ppc.o

wasppc fails with:
Error: invalid option -fr
```

The wasmps and the wasaxp assembler have the same failure as wasppc:
```
cd bld/as/mps/test
$ owcc -march=mps  hello.asm -c -o hello_mips.o
wasmps fails with:
Error: invalid option -fr
```
```
cd bld/as/axp/test
owcc -march=axp  fact.asm -c -o fact_axp.o

wasaxp fails with:
Error: invalid option -fr
```

**Comment by jmalak** [comment]
You like do statements about something is wrong without any info.
By example you talk about "--help is not listed" it is non-sense any OW tools doesn't use such option.
-? -h options is OW pain, it is sometimes partialy implemented or not implemented at all.
But this issue has lowest priority for me.
Only issue with -fr option is bug and should be fixed.

**Comment by jmalak** [comment]
It is fixed now by commit eda710ef8fde1fb8c02f1bf1d99b4e3cc215066f

**Comment by winspool** [comment]
The risc assembler still fail...
